R/DMMParam.R

```diff
@@ -0,0 +1,203 @@
+#' 
+#' If we have microbiota data (for example the Twins data set in the DMM package), the suitable algorithm can be DMM.
+#' To modify an existing DMMParam object \code{x},
+#' users can simply call \code{x[[i]]} or \code{x[[i]] <- value} where \code{i} is any argument used in the constructor.
+#'
+#' If \code{k=NULL}, the cluster search range will be 1:3.
+#' The default for the fit function is \code{laplace}.
+#' 
+#' @return 
+#' The \code{DMMParam} constructor will return a \linkS4class{DMMParam} object with the specified parameters.
+#'
+#' The \code{clusterRows} method will return a factor of length equal to \code{nrow(x)} containing the cluster assignments.
+#' If \code{full=TRUE}, a list is returned with \code{clusters} (the factor, as above) and \code{objects} 
+#' (a list containing \code{dmm}, the dmm object; \code{k}, the number of clusters; 
+#' \code{prob}, the array of probabilities; and \code{seed} the seed for the dmm).
+#'
+#' @examples
+#' clusterRows(iris[,1:4], DMMParam())
+#' clusterRows(iris[,1:4], DMMParam(k=2))
+#' \dontrun{
+#' fl <- system.file(package="DirichletMultinomial", "extdata", "Twins.csv")
+#'         counts <- t(as.matrix(read.csv(fl, row.names=1)))
+#'         clusterRows(counts, DMMParam(k=1:3, type="laplace"))
+#' }
+#' # Example using a SummarizedExperiment
+#' data("GlobalPatterns", package="mia")
+#' tse <-GlobalPatterns[1:1000, ]
+#' x <- assay(tsebis, "counts")
+#' dmm <- clusterRows(x, DMMParam(k=1:3))
+#'
+#' @name DMMParam-class
+#' @docType class
+#' @aliases 
+#' .defaultScalarArguments,DMMParam-method
+#' show,DMMParam-method
+NULL
+
+#' @export
+#' @rdname DMMParam-class
+setClass("DMMParam", contains="BlusterParam", slots=c(k="integer_OR_NULL", 
+                                                      type="character",
+                                                      transposed="logical",
+                                                      seed = "integer_OR_NULL"))
+
+#' @export
+#' @rdname DMMParam-class
+DMMParam <- function(k=NULL, type=NULL, transposed=FALSE, seed=NULL) {
+    # Filling in missing values with the defaults.
+    current <- list(k=k, type=type, seed=seed)
+    notpresent <- vapply(current, is.null, FALSE)
+    if (any(notpresent)) {
+        defaults <- list(k=1:3, 
+                         type="laplace", 
+                         seed=runif(1, 0, .Machine$integer.max))
+        current[notpresent] <- defaults[notpresent]
+    }
+    
+    # Formatting data
+    if (!is.null(current$k)) {
+        current$k <- as.integer(current$k)
+    }
+    if (!is.null(current$seed)) {
+        current$seed <- as.integer(current$seed)
+    }
+
+    new("DMMParam", k=current$k, type=current$type, 
+        transposed=transposed, seed=current$seed)
+}
+
+#' @export
+setMethod("show", "DMMParam", function(object) {
+    callNextMethod()
+    if (length(object@k) > 1) {
+        cat(sprintf("k: %s\n", paste(range(object@k), collapse = ":")))
```

Comment:
Why `range`? You're not showing the specific choices. Why not just use `collapse = ", "` on `object@k`? Or even `S4Vectors::coolcat` for nice truncation of large vectors?